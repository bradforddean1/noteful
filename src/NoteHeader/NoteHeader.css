.NoteHeader {
  display: flex;
  flex-direction: column;
  padding: 10px;
  border: 1px solid #000000;
}

.NoteHeader .top {
  display: flex;
  justify-content: start;
  align-content: center;
}

.NoteHeader .bottom {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.NoteHeader button {
  padding: 5px;
}
