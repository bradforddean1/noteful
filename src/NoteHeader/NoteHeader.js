import React from "react";
import "./NoteHeader.css";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import Moment from "moment";

/**
 *
 *@component
 */
function NoteHeader(props) {
  const date = new Moment(props.modDate).format("MM-DD-YYYY");
  const linkRoute = "/note/".concat(props.id);

  return (
    <div className="NoteHeader">
      <div className="top">
        <h2>
          <Link to={linkRoute}>{props.name}</Link>
        </h2>
      </div>
      <div className="bottom">
        <span>Date modified on {date}</span>
        <button
          onClick={() => {
            return;
          }}
        >
          Delete Note
        </button>
      </div>
    </div>
  );
}

NoteHeader.propTypes = {
  name: PropTypes.string,
  modDateate: PropTypes.string,
  id: PropTypes.string,
};

NoteHeader.defaultProps = {
  name: "_",
};

export default NoteHeader;
