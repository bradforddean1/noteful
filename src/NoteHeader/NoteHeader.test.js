import React from "react";
import ReactDOM from "react-dom";
import NoteHeader from "./NoteHeader";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("NoteHeader Component", () => {
  // Smoke test
  test("Renders", () => {
    const div = document.createElement("div");
    ReactDOM.render(<NoteHeader id="id" deleteNote={() => null} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  // Snapshot tests
  test("Renders default state", () => {
    const wrapper = shallow(<NoteHeader id="id" deleteNote={() => null} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test("Renders state with NoteHeader name", () => {
    const wrapper = shallow(
      <NoteHeader
        id="id"
        deleteNote={() => null}
        modDate="01/01/2020"
        name="Dogs"
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
