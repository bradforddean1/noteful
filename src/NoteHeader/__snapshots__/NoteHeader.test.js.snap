// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteHeader Component Renders default state 1`] = `
<div
  className="NoteHeader"
>
  <div
    className="top"
  >
    <h2>
      <Link
        to="/note/id"
      >
        _
      </Link>
    </h2>
  </div>
  <div
    className="bottom"
  >
    <span>
      Date modified on 
      Invalid date
    </span>
    <button
      onClick={[Function]}
    >
      Delete Note
    </button>
  </div>
</div>
`;

exports[`NoteHeader Component Renders state with NoteHeader name 1`] = `
<div
  className="NoteHeader"
>
  <div
    className="top"
  >
    <h2>
      <Link
        to="/note/id"
      >
        Dogs
      </Link>
    </h2>
  </div>
  <div
    className="bottom"
  >
    <span>
      Date modified on 
      01-01-2020
    </span>
    <button
      onClick={[Function]}
    >
      Delete Note
    </button>
  </div>
</div>
`;
